app/scripts/app.js

```diff
@@ -56,12 +61,17 @@ angular.module('yourStlCourts').config(function($stateProvider, $urlRouterProvid
         court: {value: undefined}
       },
       resolve: {
-        court: function ($stateParams,Courts) {
-          if ($stateParams.courtId != ""){
-            return Courts.findById($stateParams.courtId)
-          }else{
-            return null;
-          };
+        court: function ($stateParams,Courts,Error) {
+          if ($stateParams.courtId == "") {
+            throw Error.errorObject(ErrorCode.BAD_REQUEST, "No Court was found with the url you provided.");
+          } else {
+            return (Courts.findById($stateParams.courtId)
+              .then(function (court) {
+                return court;
+              },function(){
```

Comment:
This is unnecessary. When you're only going to return the thing the promise returns already, you don't need ".then"....consider:

```
return Courts.findById($stateParams.courtId).catch(function() {
  throw Errors.makeError(Errors.ERROR_CODE.NOT_FOUND, "No Court was found with the url you provided.");
});
```